### PR TITLE
Add ironic-standalone chart

### PR DESCRIPTION
**Pull request by MaxRink**
This adds our version of ironic-standalone to this helm repo

### CODE REVIEW

File: charts/ironic-standalone/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  - name: jakub-d
+    email: jakub.dlugolecki@gmail.com
+  - name: MaxRink
+    email: maximilian.rink@telekom.de
+  dependencies:
+  - name: mariadb
+    version: "9.x.x"
+    repository: "https://charts.bitnami.com/bitnami"
+    condition: mariadb.enabled
+  - name: rabbitmq
```
Comment (dtantsur) [inline charts/ironic-standalone/Chart.yaml:18]: Do you really need to use rabbitmq?

File: charts/ironic-standalone/README.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+Keystone Authentication (Optional)
+==================================
+
+This chart deploys only Openstack Ironic by default.
+You can enable Keystone service by providing extra configuration parameters in the deployment:
+
+```
+keystone:
+  enabled: true
+  admin_password: ADMIN-PASSWORD
```
Comment (dtantsur) [inline charts/ironic-standalone/README.md:100]: This looks like a pretty dated keystone configuration. I think (not sure) admin_password is no longer a thing.

File: charts/ironic-standalone/README.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+It uses `subPath` functionality to create sub directories on the existing claim.
+The existing claim name must be defined in two places: `persistentVolumeClaimName` and `mysql.persistence.existingClaim`.
+
+One can upload disk images using HTTP PUT request:
+
+```
+curl -T diskimage.qcow2 http://<httpboot_externalIP>
+```
+
+Note on tftpd
```
Comment (dtantsur) [inline charts/ironic-standalone/README.md:141]: Better use dnsmasq

**Comment by dtantsur** [comment]
I'm not sure I'm aware of the goal of this project, but this looks like a cargo-cult from another place, which does have a few dated bits. I wonder why not just lunch ironic-image from Metal3. The ironic team here cannot maintain yet another source of truth in Metal3.

File: charts/ironic-standalone/Chart.yaml (hunk 1 above)
Comment (MaxRink) [reply charts/ironic-standalone/Chart.yaml:18]: Yes, if you want to scale your ironic instances for HA

**Comment by MaxRink** [comment]
> I wonder why not just lunch ironic-image from Metal3.

Because monolithic images are a huge antipattern.
The metal3 image bunces ironic, dns and DB in one image, which is something you should not do, ever.

File: charts/ironic-standalone/README.md (hunk 3 above)
Comment (MaxRink) [reply charts/ironic-standalone/README.md:141]: might be, but we actually never used PXE boot so hwe havent tested that

File: charts/ironic-standalone/README.md (hunk 2 above)
Comment (MaxRink) [reply charts/ironic-standalone/README.md:100]: Probably, but we never used keystone auth so far, as we dont run the necassesary components for that, just basic auth, this is mostly in here to

**Comment by dtantsur** [comment]
> The metal3 image bunces ironic, dns and DB in one image, which is something you should not do, ever.

I don't disagree with that, but then you should direct your work towards ironic-image (I'd very happily split this stuff out).

File: charts/ironic-standalone/Chart.yaml (hunk 1 above)
Comment (dtantsur) [reply charts/ironic-standalone/Chart.yaml:18]: This is only correct for ironic-inspector, only if you need it in active/active HA, and only until we fix it upstream.

File: charts/ironic-standalone/README.md (hunk 2 above)
Comment (dtantsur) [reply charts/ironic-standalone/README.md:100]: This is keystone configuration, not basic auth.

**Comment by dtantsur** [comment]
Also have you cleared up https://github.com/jakub-d/openstack-ironic-standalone/issues/12?

**Comment by zaneb** [comment]
Running Ironic with RabbitMQ is many things, but it is not Metal³. If that is the approach you want to go for then I don't think it belongs in an upstream repo, the potential for confusion is too great.

**Comment by dtantsur** [comment]
Yep. And if you'd like to align with Metal3 upstream, we need to somehow share configuration between ironic-image and this work.

**Comment by MaxRink** [comment]
> > The metal3 image bunces ironic, dns and DB in one image, which is something you should not do, ever.
> 
> I don't disagree with that, but then you should direct your work towards ironic-image (I'd very happily split this stuff out).

Will do, illopen up a discussion item on tze image repo later on

**Pull request closed without merging**